src/System.Buffers.Primitives/System/Buffers/Pooling/MemoryPool.cs

```diff
@@ -3,23 +3,25 @@
 
 namespace System.Buffers
 {
-    public abstract class BufferPool : IDisposable
+    public abstract class MemoryPool<T> : IDisposable
     {
-        public static BufferPool Default => Internal.ManagedBufferPool.Shared;
+        public static MemoryPool<T> Default => Internal.ArrayMemoryPool<T>.Shared;
 
-        public abstract OwnedMemory<byte> Rent(int minimumBufferSize);
+        public abstract OwnedMemory<T> Rent(int minimumBufferSize);
+
+        public abstract void Return(OwnedMemory<T> buffer);
```

Comment:
Do we need `Return` at all if we have `Release` in `OwnedMemory<T>`?

Reply:
Yea, this was a point of contention before.

Reply:
I added the method to allow pooling of the OwnedMemory instances.